### ISSUE TITLE
Facebook Pixel Extension Issue 

### ISSUE THREAD

**Issue by adedolaisaac** [facebookincubator/Facebook-For-OpenCart#75]
I have now spent over 10 hours trying to figure this out. I installed the 2.1.2 version on a 3.0.2, the installation worked perfectly. But to my biggest surprise facebook pixel heler didn't detect the pixel installation on my server. I have done series of troubleshooting myself.  See below:

* Checked the source code directly using view source option on the brower => No pixel script installed
* Monitored facebook real-time pixel test environment on facebook pixel manager => Nothing fired
* Checked header.twig => No pixel added either, and file is not modified neither
* Checked the ocmod modification file => Seems perfect
* Done several re-installation => Still to noavail

Also in the Facebook Extension module area, I got "Error Not Found" after installation, couldn't figure out why that also happened, because my chmod permission seems okay the folders are on 775, in reference to admin/controller/extension/facebook*.php files. The files are in 644 which is normal.

So please could you advice me on what should be done? I have deadline and need to get this pixel to start firing like a roooocket...!! 👍 Thanks.

**Comment by shue249** [comment]
@adedolaisaac Thank you for your feedback! We are sorry that the experience is not as ideal and it took you time to solve this.

2 suggestions.

1. Can you look at this video, did you perform this step? https://drive.google.com/open?id=1JPXQ0mS1pGk2Bat9RV02vYNdbRfCLP9r

2. Can you check if these files are existing on your server after installing the plugin?
![image](https://user-images.githubusercontent.com/11957585/49630808-b1998980-fa2a-11e8-9907-1c9fb5401f89.png)

3. Are the admin, catalog and system having 775 permissions in recursive mode?

**Comment by adedolaisaac** [comment]
Yes, I followed the steps in the video. I just checked, all my folders are in 755, I wonder why I didn't notice them, so do I need to reinstall it again after changing the permission 775 or just flush the cache?

**Comment by shue249** [comment]
@adedolaisaac I like to learn more on your situation. I believe that you should be able to see the Facebook Ads Extension plugin on your menu bar, but you are unable to see the contents of that module, ie it shows out as Page not found. Is that correct?
![image](https://user-images.githubusercontent.com/11957585/49633298-d80ff280-fa33-11e8-938f-d36fa58c738b.png)

You need to re-install the plugin again (need to delete away the existing plugin and install again), after changing the permission to 775. 

To give full context here. We have some new files from the plugin that we need to copy over to your webserver. If the folder permission is not writeable, the files will not be able to be copied over.

**Comment by adedolaisaac** [comment]
Thanks for the swift response and tailored step by step information. However, I do not have any issue accessing this part of the extension. 

1) As you can see in the screenshot below, the only part not visible is the cookies section. 
![fpage](https://user-images.githubusercontent.com/33562400/49633930-4214a180-fa0c-11e8-8ec2-e3917dec6a69.png)

2) Here is where I have the issue. Extensions > Modules > Facebook Ads Extension > Edit
![fpage2](https://user-images.githubusercontent.com/33562400/49634010-8bfd8780-fa0c-11e8-919e-1bf1cf00aa55.png)

3) Here is what the page looks like.
![fpage3](https://user-images.githubusercontent.com/33562400/49634137-e26ac600-fa0c-11e8-9deb-c45c69dbfc9f.png)

And secondly, after I changed the permissions to 775, access to the admin area became inaccessible as all the links in the back-end area were redirecting to 404 front-end area. This error got resolved immediately after I returned the permission to 755. So what do you think should be done?

**Comment by adedolaisaac** [comment]
I just checked the modification error log, and this is what I got.

`----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/admin/view/template/common/dashboard.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/admin/controller/common/menu.php
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/admin/view/template/common/menu.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/admin/view/template/catalog/product_list.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/catalog/view/theme/*/template/common/header.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/catalog/view/theme/*/template/product/product.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/catalog/view/theme/*/template/common/footer.tpl

MOD: Facebook Ads Extension
NOT FOUND!
CODE: </footer>
FILE: catalog/view/theme/theme-name/template/common/footer.twig

MOD: Facebook Ads Extension
NOT FOUND!
CODE: </footer>
FILE: catalog/view/theme/theme-name-mobile/template/common/footer.twig
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/catalog/controller/common/seo_url.php`

I think issue still resolves to the permission

**Comment by adedolaisaac** [comment]
I just checked the modification error log, and this is what I got.

----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/admin/view/template/common/dashboard.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/admin/controller/common/menu.php
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/admin/view/template/common/menu.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/admin/view/template/catalog/product_list.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/catalog/view/theme/*/template/common/header.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/catalog/view/theme/*/template/product/product.tpl
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/catalog/view/theme/*/template/common/footer.tpl

MOD: Facebook Ads Extension
NOT FOUND!
CODE: </footer>
FILE: catalog/view/theme/theme-name/template/common/footer.twig

MOD: Facebook Ads Extension
NOT FOUND!
CODE: </footer>
FILE: catalog/view/theme/theme-name-mobile/template/common/footer.twig
----------------------------------------------------------------
MOD: Facebook Ads Extension
MISSING FILE!
/home/casorashop/public_html/catalog/controller/common/seo_url.php

I think issue still resolves to the permission

**Comment by shue249** [comment]
@adedolaisaac Ok, I am seeing more context with your screenshot.

1. I see that you have completed the Facebook Ads Extension setup. This means the installation is done and you should be good. There is no need to click on the Edit button of the modification as we do not have any features there.

2. The errors from the modifications should be fine as they are intended for the OpenCart v2.x. We developed the plugin to be suitable for both 3.x and 2.x. Can you share with me your website URL?

3. I see 1 problem is that that you are unable to see the option to disable the cookie bar. Can you enable the console log of your webpage and do a refresh on the Facebook Ads Extension module. I like to see if there are any errors reported out.

**Comment by adedolaisaac** [comment]
I just checked the modification error log, and this is what I got.


> MOD: Facebook Ads Extension
> MISSING FILE!
> /home/[**]/public_html/admin/view/template/common/dashboard.tpl
> 
> MOD: Facebook Ads Extension
> MISSING FILE!
> /home/[**]/public_html/admin/controller/common/menu.php
> 
> MOD: Facebook Ads Extension
> MISSING FILE!
> /home/[**]/public_html/admin/view/template/common/menu.tpl
> 
> MOD: Facebook Ads Extension
> MISSING FILE!
> /home/[**]/public_html/admin/view/template/catalog/product_list.tpl
> 
> MOD: Facebook Ads Extension
> MISSING FILE!
> /home/[**]/public_html/catalog/view/theme/*/template/common/header.tpl
> 
> MOD: Facebook Ads Extension
> MISSING FILE!
> /home/[**]/public_html/catalog/view/theme/*/template/product/product.tpl
> 
> MOD: Facebook Ads Extension
> MISSING FILE!
> /home/[**]/public_html/catalog/view/theme/*/template/common/footer.tpl
> 
> MOD: Facebook Ads Extension
> NOT FOUND!
> CODE: </footer>
> FILE: catalog/view/theme/theme-name/template/common/footer.twig
> 
> MOD: Facebook Ads Extension
> NOT FOUND!
> CODE: </footer>
> FILE: catalog/view/theme/theme-name-mobile/template/common/footer.twig
> 
> MOD: Facebook Ads Extension
> MISSING FILE!
> /home/[**]/public_html/catalog/controller/common/seo_url.php

I think issue still resolves to the permission

**Comment by adedolaisaac** [comment]
@shue249 the website is casorashop.com

**Comment by adedolaisaac** [comment]
@shue249 From the error log on with request on load. I got this error

> Warning: html_entity_decode(): charset `UTF-8;' not supported, assuming utf-8 in /home/[**]/public_html/system/library/facebookcommonutils.php on line 159

Here is the code of the function where line 159 precedes

159 ~> html_entity_decode($text),
```
  public function getProperFormattedString($text) {
    if ((bool)$text) {
      return trim(strip_tags(html_entity_decode(
        html_entity_decode($text),
        ENT_QUOTES | ENT_COMPAT,
        'UTF-8')));
    } else {
      return '';
    }
  }
```
My php.ini is properly set at "UFT-8" 
default_charset	= UTF-8;

Please let me know your suggestion.

**Comment by adedolaisaac** [comment]
@shue249 Hello again, I fixed the error by removing the semicolon from the end of the line `default_charset = UTF-8; `in my .htaccess, user.ini and php.ini files. But Facebook Extension issue still persists.

**Comment by shue249** [comment]
@adedolaisaac I took a look at your website. The problem is that the facebook pixel did not get installed correctly on your website.

**Comment by shue249** [comment]
![image](https://user-images.githubusercontent.com/11957585/49656987-930eaf00-fa79-11e8-82d5-3a33e4415e4b.png)

**Comment by adedolaisaac** [comment]
I previously attached a screenshot of the log in my previous thread, see below again for reference. In addition, I was thinking about fixing the bugs in the codes but would take a while. There are some files that are suppose to be .twig instead of .tpl, and I found some invalid request as well. According to my observation there is a missing </footer> tag in `/theme/theme-name/template/common/footer.twig` I found this tag in `catalog/view/theme/theme-name/template/common/menu.twig`. There was an invalid request too on `/home/[**]/public_html/catalog/controller/common/seo_url.php` where the valid request should be `/home/[**]/public_html/catalog/controller/**startup**/seo_url.php`. Please let me know what you think?

> 
> 
> MOD: Facebook Ads Extension
> MISSING FILE!
> /public_html/admin/view/template/common/dashboard.tpl
> MOD: Facebook Ads Extension
> MISSING FILE!
> /public_html/admin/controller/common/menu.php
> MOD: Facebook Ads Extension
> MISSING FILE!
> /public_html/admin/view/template/common/menu.tpl
> MOD: Facebook Ads Extension
> MISSING FILE!
> /public_html/admin/view/template/catalog/product_list.tpl
> MOD: Facebook Ads Extension
> MISSING FILE!
> /public_html/catalog/view/theme/*/template/common/header.tpl
> MOD: Facebook Ads Extension
> MISSING FILE!
> /public_html/catalog/view/theme/*/template/product/product.tpl
> MOD: Facebook Ads Extension
> MISSING FILE!
> /public_html/catalog/view/theme/*/template/common/footer.tpl
> MOD: Facebook Ads Extension
> NOT FOUND!
> CODE: 
> FILE: catalog/view/theme/theme-name/template/common/footer.twig
> MOD: Facebook Ads Extension
> NOT FOUND!
> CODE: 
> FILE: catalog/view/theme/theme-name-mobile/template/common/footer.twig
> MOD: Facebook Ads Extension
> MISSING FILE!
>  /public_html/catalog/controller/common/seo_url.php
>

**Comment by shue249** [comment]
@adedolaisaac I want to follow up with you on this issue, as we have a new idea.

For OpenCart v3.0.2.0, you need to disable the theme cache for the UI modifications to take effect. Can you look at this screenshot and see if it helps you. https://drive.google.com/open?id=1bY-bworYxX36b88HDvFW0_32C3Wtq_Tm

**Comment by cadmiumc** [comment]
@shue249 Hi, Thanks for your effort to the extension!  I have a similar issue to adedolaisaac. 
Aim: Everything worked smoothly but cannot disable the cookies setting.
OC Version: 3.0.3.1
Plugin Version: 2.1.3 (firstly installed Plugin Version: 2.0.3, switch to the lastest version now.)
- I have installed the extension correctly (I think so, because I dont have the error log raised in modification.), but the cookies section is not visible.  
- I have read and tried the method of #98 but it doesnt worked. 
- I also tried your manual method in #20. While the file loaction is not exist, for "system/storage/modification/catalog/view/theme/journal2/template/common"
- I can only go to "system/storage/modification", and nothings in there.
